test/uitest.c

```diff
@@ -59,9 +59,11 @@ static int test_old()
     char pass[16];
     int ok = 0;
 
-    if ((ui_method =
-         UI_UTIL_wrap_read_pem_callback(test_pem_password_cb, 0)) == NULL
-        || (ui = UI_new_method(ui_method)) == NULL)
+    ui_method = UI_UTIL_wrap_read_pem_callback(test_pem_password_cb, 0);
+    if (!TEST_ptr(ui_method))
+        goto err;
+    ui = UI_new_method(ui_method);
+    if (!TEST_ptr(ui))
```

Comment:
There are more places in this PR where the same line of thinking could be applied